src/include/miopen/miopen_internal.h

```diff
@@ -104,6 +106,23 @@ MIOPEN_EXPORT miopenStatus_t miopenGetConvolutionFindMode(
 
 /* End of Find Mode API */
 
+/*! @brief Get extra workspace for CK's backward weight kernel.
+ *
+ * @param miopenAlphaBetaCase_t type of alpha beta case
+ * @param  miopenDataType_t data type
+ * @param  size_t input channel
+ * @param  size_t output channel
+ * @param  size_t output tensor length
+ * @param  size_t buffer_size for CK Backward weights work space
+ */
+extern "C" miopenStatus_t
+miopenConvolutionCKBackwardWeightsGetWorkSpaceSize(const miopenAlphaBetaCase_t ab_case,
+                                                   miopenDataType_t data_type,
+                                                   size_t C,
+                                                   size_t K,
+                                                   size_t output_tensor_size,
```

Comment:
So, yes, it will be more complex because of our layouts, but it's also future proof. 
Secondly, you're asking the user to provide `output_tensor_size` which is more error prone to the user. It depends on the layout (packed vs non-packed) and can be confusing (are you asking bytes or elements?).

Reply:
ok. I am asking elements. Will compute bytes based on `data_type`

Reply:
yes, I know you're asking for num elements, but the main point being, make the interface easier and future proof by taking descriptor pointers instead of `C`, `K` etc. What if you need one of `G, D, H, W` in the future?

Reply:
good point!